RIOT-OS/RIOT
# cpu/esp32: download Espressif SDK ESP-IDF as package

### Contribution description

This PR adds the ESP32 vendor SDK `ESP-IDF` as package.

RIOT-OS uses the Espressif's vendor SDK `ESP-IDF` to support the ESP32 CPU. The SDK contains some closed-source precompiled libraries provided by the vendor. Up to now, it was necessary to manually install the SDK and set up the environment. With this PR, the SDK is downloaded as a package at build time.

Since the SDK and its precompiled libraries are provided as separate GIT repositories, two different packages are defined and pulled separately:
- SDK without precompiled vendor libraries (`esp32_sdk`)
- precompiled SDK vendor libraries (`esp32_sdk_libs`)

There are two reason for defining separate packages:
1. RIOT packages don't support to clone GIT repositories recursively
2. ESP-IDF pulls a lot of other GIT that are not needed repositories when it is cloned recursively

Using SDK as package requires about to download 450 MByte in clean environment, not sure whether this is acceptable:
```
du -hs build/pkg/esp32_sdk*
259M	build/pkg/esp32_sdk
188M	build/pkg/esp32_sdk_libs
```
But in long-term, the SDK could be removed from `riotdocker` and the SDK follows the RIOT versioning.

### Testing procedure

Tested with
```
ESP32_SDK_DIR=/dev/null USEMODULE=esp_log_startup \
BOARD=esp32-wroom-32 make -C examples/gnrc_networking flash term
```
and verify the SDK version:
```
ESP-IDF SDK Version v3.1-beta1-41-gf198339ec0
```
Setting `ESP32_SDK_DIR=/dev/null` is only for testing and just to make sure that no evaluation of `ESP32_SDK_DIR` in make is affected by the environment variable which would be set to the SDK for people who followed the set up instructions before this change.

### Issues/PRs references

Follow-up to PR #16425
